java/src/jmri/jmrit/display/layoutEditor/LayoutTrack.java

```diff
@@ -71,7 +72,11 @@
     /**
      * constructor method
      */
-    public LayoutTrack() {
+    public LayoutTrack(@Nonnull String ident, @Nonnull Point2D c, @Nonnull LayoutEditor layoutEditor) {
+        this.instance = this;
```

Comment:
Why are you maintaining this reference? Anywhere the use of this reference is valid, the use of "this" is also valid.

Reply:
Not my code… don't know why it's there. I reverse inherited it from what are now subclasses of LayoutTrack.
Looking at the (30) usages…

```
            popup.add(new AbstractAction(Bundle.getMessage("ButtonEdit")) {
                @Override
                public void actionPerformed(ActionEvent e) {
                    editLayoutSlip((LayoutSlip) instance);
                }
            });

```
If you change instance to this you get a compile error "incompatible types: <anonymous AbstractAction> cannot be converted to LayoutSlip.". So apparently in this case 'this' refers to the AbstractAction. One solution would be to use a local variable:
```
            LayoutSlip thisLayoutSlip = this;
            popup.add(new AbstractAction(Bundle.getMessage("ButtonEdit")) {
                @Override
                public void actionPerformed(ActionEvent e) {
                    editLayoutSlip(thisLayoutSlip);
                }
            });
```
I'm guessing the original developer thought that one class variable was better than 30 locals… Do you think I should change them all?

Reply:
What you're looking for in the inner class is a "qualified this".  Try
```
LayoutSlip.this
```
See https://stackoverflow.com/questions/2858604/accessing-the-containing-class-of-an-inner-class-in-java